datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java

```diff
@@ -825,8 +734,33 @@ private void configureDatasetResolvers(final RuntimeWiring.Builder builder) {
                 .dataFetcher("schemaMetadata", new AuthenticatedResolver<>(
                     new AspectResolver())
                 )
+                .dataFetcher("tags", new AuthenticatedResolver<>(
```

Comment:
One reason we've been hesitant to do this is that we want to batch fetch all the aspects for an entity at once - previously we used to have 2 microservices so each resolver ended up in a separate network request. 

We had continued to use this approach to make it conducive to splitting again later on if we needed to... 

So this change just trades off easily splitting GMS from GraphQL server for ability to fetch version of each aspect

Reply:
cc @gabe-lyons for visibility

Reply:
Yeah wasn't sure whether they should be batched together or not. Very much possible to pivot to batching them all together, but wanted to keep in line with the current design. Happy to have a design session to get this the way we want it or revisit and come back with some proposals.

Reply:
I think we should opt for code simplicity here over a potential performance optimization down the line.